src/drivers/intel/cavs/clk.c

```diff
@@ -139,11 +138,17 @@ uint32_t clock_set_freq(int clock, uint32_t hz)
 	notify_data.old_freq = clk_pdata->clk[clock].freq;
 	notify_data.old_ticks_per_msec = clk_pdata->clk[clock].ticks_per_msec;
 
-	/* atomic context for chaning clocks */
+	/* atomic context for changing clocks */
 	spin_lock_irq(&clk_pdata->clk[clock].lock, flags);
 
 	switch (clock) {
-	case CLK_CPU:
+	case CLK_CPU(0):
+	case CLK_CPU(1):
+#if defined(CONFIG_CANNONLAKE) || defined(CONFIG_ICELAKE) || \
```

Comment:
Number of CPUs (cores) should be defined in one of the platform configuration headers and be respected here during compilation. CONFIG_<platform> should disappear, wherever possible, from the cavs platform code. Compilation should be feature, not platform oriented wherever possible.